x-pack/plugins/synthetics/server/legacy_uptime/lib/saved_objects/migrations/monitors/8.8.0.ts

```diff
@@ -118,3 +126,102 @@ const omitZipUrlFields = (fields: BrowserFields) => {
 
   return formatSecrets(validationResult.decodedMonitor);
 };
+
+const updateThrottlingFields = (
+  doc: SavedObjectUnsanitizedDoc<
+    SyntheticsMonitorWithSecrets &
+      Partial<{
+        [LegacyConfigKey.THROTTLING_CONFIG]: string;
+        [LegacyConfigKey.IS_THROTTLING_ENABLED]: boolean;
+        [LegacyConfigKey.DOWNLOAD_SPEED]: string;
+        [LegacyConfigKey.UPLOAD_SPEED]: string;
+        [LegacyConfigKey.LATENCY]: string;
+      }>
+  >,
+  logger: SavedObjectMigrationContext
+) => {
+  try {
+    const { attributes } = doc;
+
+    const migrated = {
+      ...doc,
+      attributes: {
+        ...doc.attributes,
+        [ConfigKey.CONFIG_HASH]: '',
+      },
+    };
+
+    const isThrottlingEnabled = attributes[LegacyConfigKey.IS_THROTTLING_ENABLED];
+    if (isThrottlingEnabled) {
+      const download = String(attributes[LegacyConfigKey.DOWNLOAD_SPEED])!;
+      const upload = String(attributes[LegacyConfigKey.UPLOAD_SPEED])!;
+      const latency = String(attributes[LegacyConfigKey.LATENCY])!;
+
+      // @ts-ignore
```

Comment:
Without having looked deeper at the code, I find it strange that you have to use ts-ignore. Is there a way we can improve the types to avoid it? If not, it would be preferable to force cast because ts-ignore will hide any type errors.

Reply:
@shahzad31 Can you perhaps cast the attributes to `BrowserFields` since this part of the migration exists in a conditional that already determines that it's a browser monitor?